Compilation errors when building pipe viewer

```
[100%] Linking CXX executable Argos_dumper
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:12549:21: error: fallthrough annotation in unreachable code [-Werror,-Wunreachable-code-fallthrough]
                    CYTHON_FALLTHROUGH;
                    ^
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:430:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
      #define CYTHON_FALLTHROUGH [[fallthrough]]
                                 ^
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:12560:21: error: fallthrough annotation in unreachable code [-Werror,-Wunreachable-code-fallthrough]
                    CYTHON_FALLTHROUGH;
                    ^
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:430:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
      #define CYTHON_FALLTHROUGH [[fallthrough]]
                                 ^
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:13719:21[100%] Built target Argos_dumper
: error: fallthrough annotation in unreachable code [-Werror,-Wunreachable-code-fallthrough]
                    CYTHON_FALLTHROUGH;
                    ^
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:430:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
      #define CYTHON_FALLTHROUGH [[fallthrough]]
                                 ^
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:13730:21: error: fallthrough annotation in unreachable code [-Werror,-Wunreachable-code-fallthrough]
                    CYTHON_FALLTHROUGH;
                    ^
/Users/zen/Sparta/map/helios/pipeViewer/pipe_view/core/src/core.cpp:430:34: note: expanded from macro 'CYTHON_FALLTHROUGH'
      #define CYTHON_FALLTHROUGH [[fallthrough]]
```

--- a/helios/pipeViewer/pipe_view/core/setup.py
+++ b/helios/pipeViewer/pipe_view/core/setup.py
@@ -103,7 +103,6 @@ compile_args += [
     '-Wno-deprecated-declarations',
     '-Wno-strict-aliasing',
     '-Wall',
-    '-Werror',
     '-Wpedantic'
 ]
 

--- a/sparta/sparta/functional/RegisterSet.hpp
+++ b/sparta/sparta/functional/RegisterSet.hpp
@@ -637,7 +637,7 @@ public:
     /*!
      * \brief Dumps the table of banks and registers
      */
-    void dumpBanks(std::ostream& out) {
+    void dumpBanks(std::ostream& out) const {
         banks_.dump(out);
     }
 
@@ -1116,4 +1116,3 @@ inline std::ostream &operator<<(std::ostream &o, const sparta::RegisterSet *rs)
     }
     return o;
 }
-

--- a/sparta/sparta/simulation/Parameter.hpp
+++ b/sparta/sparta/simulation/Parameter.hpp
@@ -1573,15 +1573,21 @@ namespace sparta
         /*!
          * \brief Assigns the specified value to this parameter
          * \todo Perform independent parameter validation and throw exception
-         * if failed.
+         *       if failed.
          * \post Write count is incremented and read count reset
+         *
+         * \note If the user is initializing the parameters using a
+         *       final config, then parameters assignments are not
+         *       allocated.  The user must use the `override*`
+         *       methods.  This does _not_ apply to parameters whose
+         *       visibility is hidden
          */
         void operator=(const ValueType& v) {
             checkModificationPermission_();
             // if the simulator was setup using --read-final-config
             // we do not allow the simulator to override values
             // so just return out and warn that we are skipping this.
-            if (usingFinalConfig_())
+            if (usingFinalConfig_() && (false == isHidden()))
             {
                 static bool been_warned = false;
                 if(!been_warned) {
@@ -1662,26 +1668,6 @@ namespace sparta
             incrementWriteCount_();
 
             val_ = def_val_;
-        }
-
-        virtual void setValueFromStringImpl_(const std::string& str, bool poke=false) override final {
-            Parameter::template setValueFromString_<ValueType, ValueType>(str, poke);
-        }
-
-        virtual void setValueFromStringVectorImpl_(const std::vector<std::string>& vec, bool poke=false) override final {
-            Parameter::template setValueFromStringVector_<ValueType, ValueType>(vec, poke);
-        }
-
-        virtual void setItemValueFromStringImpl_(const std::vector<uint32_t>& indices,
-                                                 const std::string& str) override final {
-            if(indices.size() == 0){
-                // No indices given. This must be a non-vector type
-                Parameter::template setValueFromString_<ValueType, ValueType>(str);
-            }else{
-                // handle if this is a vector
-                const bool incr_write_count = true;
-                setVectorItemValueFromString_(val_, indices, 0, str, incr_write_count);
-            }
         }
 
         virtual bool equals(const ParameterBase &other) override final {
@@ -1769,6 +1755,17 @@ namespace sparta
             return Parameter::template operator_insert_<U, ValueType>(e);
         }
 
+        /*!
+         * \brief Query if this parameter is safe to be displayed via prints, dumps.
+         * A parameter should not be displayed if it is HIDDEN and the subtree containing
+         * this parameter is already locked.
+         */
+        virtual bool isVisibilityAllowed() const override{
+            return ((param_attr_ == ParameterAttribute::HIDDEN) and
+                    (this->areParametersLocked_())) ? false : true;
+        }
+
+    protected:
 
         template <class U, class C1>
         typename std::enable_if<is_vector<C1>::value, Parameter<ValueType>&>::type
@@ -1779,16 +1776,25 @@ namespace sparta
             return *this;
         }
 
-        /*!
-         * \brief Query if this parameter is safe to be displayed via prints, dumps.
-         * A parameter should not be displayed if it is HIDDEN and the subtree containing
-         * this parameter is already locked.
-         */
-        virtual bool isVisibilityAllowed() const override{
-            return ((param_attr_ == ParameterAttribute::HIDDEN) and
-                    (this->areParametersLocked_())) ? false : true;
+        virtual void setValueFromStringImpl_(const std::string& str, bool poke=false) override final {
+            Parameter::template setValueFromString_<ValueType, ValueType>(str, poke);
         }
-    protected:
+
+        virtual void setValueFromStringVectorImpl_(const std::vector<std::string>& vec, bool poke=false) override final {
+            Parameter::template setValueFromStringVector_<ValueType, ValueType>(vec, poke);
+        }
+
+        virtual void setItemValueFromStringImpl_(const std::vector<uint32_t>& indices,
+                                                 const std::string& str) override final {
+            if(indices.size() == 0){
+                // No indices given. This must be a non-vector type
+                Parameter::template setValueFromString_<ValueType, ValueType>(str);
+            }else{
+                // handle if this is a vector
+                const bool incr_write_count = true;
+                setVectorItemValueFromString_(val_, indices, 0, str, incr_write_count);
+            }
+        }
 
         //! Iternal getValue_ wrapper which does not increment the read counter
         const ValueType& getValue_() const {

